### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by ichala**
**in this pr i add the following :** 

- Add React Redux .
- Structure myapplication files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files. 
- Write the book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux) 
- Create the reducers and actions for books and categories

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+// Action Creators
+export function createBook(book) {
+  return { type: CREATE, payload: book };
+}
+
+export function deleteBook(id) {
+  return { type: DELETE, payload: id };
+}
```
Comment (CollinsTatang) [inline src/redux/books/books.js:22]: - [x] Please users find it difficult to `Add New Book` to the store, please try to fix it. 
- [x] Please users find it difficult to `Remove Book` from the store, please try to fix it too.

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @ichala, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (ichala) [reply src/redux/books/books.js:22]: @CollinsTatang

 **can you please explain more what the required changes? what do you mean by ' Please users find it difficult ' ??**
I think you have a typo or smthg

Comment (CollinsTatang) [reply src/redux/books/books.js:22]: Ok
Sorry about the error 
let me reach out for correction

Comment (CollinsTatang) [reply src/redux/books/books.js:22]: Hi @ichala 
I am very sorry the problem will be addressed soon.

**Review by mikethreels — APPROVED** [review]
Hi Chala, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**